aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/manifest/CentralizedManifest.java

```diff
@@ -93,14 +93,25 @@ public void putRules(List<SamplingRule> inputs, Instant now) {
         boolean invalidate = false;
 
         Map<String, CentralizedRule> rules = this.rules;
+        List<String> inputNames = new ArrayList<>(inputs.size());
 
         for (SamplingRule i : inputs) {
             if (i.getRuleName().equals(CentralizedRule.DEFAULT_RULE_NAME)) {
                 putDefaultRule(i);
             } else {
+                inputNames.add(i.getRuleName());
                 invalidate = putCustomRule(rules, i);
             }
         }
+        // Check if any rule was removed
+        if (!invalidate) {
```

Comment:
What would happen if inputs has more than one items but the last item gets true?

Reply:
I don't understand your question. 
If we receive a set with a new rule invalidate will be true and the whole set will be rebuilt so we don't need to do anything, otherwise, we check if there is a rule that is no longer present and if that's the case these changes trigger a rebuild to remove it and reconstruct the rule priorities.